Apps/W1/HybridSL/app/src/codeunits/SLPopulateVendor1099Data.Codeunit.al

```diff
@@ -0,0 +1,767 @@
+
+namespace Microsoft.DataMigration.SL;
+
+using Microsoft.Finance.GeneralLedger.Journal;
+using Microsoft.Finance.GeneralLedger.Posting;
+using Microsoft.Finance.ReceivablesPayables;
+using Microsoft.Purchases.Payables;
+using Microsoft.Finance.GeneralLedger.Account;
+using Microsoft.Finance.VAT.Reporting;
+using Microsoft.Foundation.NoSeries;
+using Microsoft.Purchases.Vendor;
+using System.Integration;
+
+codeunit 47007 "SL Populate Vendor 1099 Data"
+{
+    EventSubscriberInstance = Manual;
+    Permissions = tabledata "IRS 1099 Form Box" = R,
+                tabledata "IRS 1099 Vendor Form Box Setup" = RIM;
+
+    var
+        DefaultPayablesAccountCode: Code[20];
+        PostingDate: Date;
+        NextYearPostingDate: Date;
+        SLCurr1099Yr: Integer;
+        SLNext1099Yr: Integer;
+        Current1099YearOpen: Boolean;
+        Next1099YearOpen: Boolean;
+        GenJournalBatchDescriptionTxt: Label 'SL Vendor 1099 Tax Journal for ', Locked = true;
+        MessageCodeAbortedTxt: Label 'ABORTED', Locked = true;
+        MessageCodeCompletedTxt: Label 'COMPLETED', Locked = true;
+        MessageCodeSkippedTxt: Label 'SKIPPED', Locked = true;
+        MessageCodeStartTxt: Label 'START', Locked = true;
+        MessageCodeProcessingTxt: Label 'PROCESSING', Locked = true;
+        NoSeriesDescriptionTxt: Label 'SL Vendor 1099', Locked = true;
+        SourceCodeTxt: Label 'GENJNL', Locked = true;
+        VendorTaxBatchNameTxt: Label 'SL1099', Locked = true;
+        VendorTaxNoSeriesTxt: Label 'VENDTAX', Locked = true;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Gen. Jnl.-Post", 'OnBeforeCode', '', false, false)]
+    local procedure OnBeforeCode(var GenJournalLine: Record "Gen. Journal Line"; var HideDialog: Boolean)
+    begin
+        HideDialog := true;
+    end;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Gen. Jnl.-Post", 'OnBeforeShowPostResultMessage', '', false, false)]
+    local procedure OnBeforeShowPostResultMessage(var GenJnlLine: Record "Gen. Journal Line"; TempJnlBatchName: Code[10]; var IsHandled: Boolean)
+    begin
+        IsHandled := true;
+    end;
+
+    trigger OnRun()
+    var
+        SLCompanyAdditionalSettings: Record "SL Company Additional Settings";
+        SLVendor1099MappingHelpers: Codeunit "SL Vendor 1099 Mapping Helpers";
+    begin
+        if not SLCompanyAdditionalSettings.GetMigrateCurrent1099YearEnabled() then
+            LogMessage(MessageCodeSkippedTxt, 'SL Vendor Current 1099 Year Data Migration is not enabled on the SL Company Migration Configuration page.');
+        if not SLCompanyAdditionalSettings.GetMigrateNext1099YearEnabled() then
+            LogMessage(MessageCodeSkippedTxt, 'SL Vendor Next 1099 Year Data Migration is not enabled on the SL Company Migration Configuration page.');
+        if not SLCompanyAdditionalSettings.GetMigrateCurrent1099YearEnabled() and not SLCompanyAdditionalSettings.GetMigrateNext1099YearEnabled() then begin
```

Comment:
I would say that this check should go first so we do not log duplicate messages.